packages/ketcher-react/src/script/editor/operations/text.ts

```diff
@@ -0,0 +1,189 @@
+/****************************************************************************
+ * Copyright 2021 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+import { BaseOperation } from './base'
+import { OperationType } from './OperationType'
+import { Text, Vec2, scale } from 'ketcher-core'
+import Restruct, { ReText } from '../../render/restruct'
+
+interface TextCreateData {
+  id?: any
+  label: string
+  position: Vec2
+  type: string
+}
+
+export class TextCreate extends BaseOperation {
+  data: TextCreateData
+  performed: boolean
+
+  constructor(id: any, label: string = '', position: Vec2, type: string) {
```

Comment:
`position` and `type` is non-optional arguments, so you can't place it after optional `label`

Reply:
refactored